src/main/java/seedu/address/model/person/LastMet.java

```diff
@@ -24,13 +23,8 @@
      * @param lastMetDate date agent last meets a client
      */
     public LastMet(String lastMetDate) {
-        if (!IS_NULL_VALUE_ALLOWED) {
-            requireNonNull(lastMetDate);
-        }
-        if (lastMetDate == null) {
-            lastMetDate = "";
-        }
 
+        requireNonNull(lastMetDate);
```

Comment:
Perhaps like CurrentPlan, this could be
```suggestion
        if (!IS_BLANK_VALUE_ALLOWED) {
            requireNonNull(lastMetDate);
        }
```